module/VuFind/src/VuFind/Config/Location/ConfigLocationInterface.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+/**
+ * Interface for configuration locations
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Hebis Verbundzentrale 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Config_Location
+ * @author   Thomas Wagener <wagener@hebis.uni-frankfurt.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+
+namespace VuFind\Config\Location;
+
+/**
+ * Interface for configuration locations
+ *
+ * This is a representation for a location of a configuration. It is based on a file or directory
+ * on the system.
+ *
+ * E.g.
+ * "/usr/local/vufind/local/config/vufind/config.ini"
+ * is the path for file configuration or
+ * "/usr/local/vufind/local/config/vufind/RecordDataFormatter"
+ * for a path for a directory configuration.
+ *
+ * The file name is the name of the file or directory that contains the configuration
+ * ("config.ini" respectively "RecordDataFormatter")
+ *
+ * The base path is the path without the filename
+ * ("/usr/local/vufind/local/config/vufind/" in both cases)
+ *
+ * The configuration name is the name for the configuration VuFind internally. Most of the time it will
+ * be the file name without extension ("config" respectively "RecordDataFormatter") but it can also be
+ * something different. E.g. when myParent.ini is the parent of config.ini it will still have the config
+ * name "config" even if that is not part of the file name.
```

Comment:
Thanks for this; it is very helpful! Here are some very minor suggested adjustments:

```suggestion
 * This is a representation for the location of a configuration. It is based on a file or directory
 * on the system.
 *
 * E.g.
 * "/usr/local/vufind/local/config/vufind/config.ini"
 * is the path for a file configuration or
 * "/usr/local/vufind/local/config/vufind/RecordDataFormatter"
 * is the path for a directory configuration.
 *
 * The file name is the name of the file or directory that contains the configuration
 * (e.g. "config.ini" or "RecordDataFormatter", respectively)
 *
 * The base path is the path without the filename
 * (e.g. "/usr/local/vufind/local/config/vufind/" in both cases)
 *
 * The configuration name is the name for the configuration used internally. Most of the time it will
 * be the file name without extension ("config" or "RecordDataFormatter", respectively) but it can also be
 * something different. E.g. when myParent.ini is the parent of config.ini it will still have the config
 * name "config" even if that is not part of the file name.
```